Sources/KituraNet/HTTP/HTTPServerResponse.swift

```diff
@@ -180,11 +156,26 @@ public class HTTPServerResponse: ServerResponse {
 
         //We don't keep the connection alive on an HTTP error
         headers["Connection"] = ["Close"]
-        let response = HTTPResponseHead(version: HTTPVersion(major: 1, minor: 1), status: status, headers: headers.httpHeaders())
+
+        execute(on: channel.eventLoop) {
+            do {
+                try self.sendResponse(channel: channel, handler: handler, status: status, withBody: withBody)
+            } catch let error {
+                fatalError("Error: \(error)")
+            }
+        }
+    }
+
+    /// Send response to the client
+    private func sendResponse(channel: Channel, handler: HTTPRequestHandler, status: HTTPResponseStatus, withBody: Bool = false) throws {
+        let response = HTTPResponseHead(version: httpVersion, status: status, headers: headers.httpHeaders())
         channel.write(handler.wrapOutboundOut(.head(response)), promise: nil)
         if withBody && buffer.readableBytes > 0 {
             channel.write(handler.wrapOutboundOut(.body(.byteBuffer(buffer))), promise: nil)
+        } else if buffer.readableBytes > 0 {
```

Comment:
This case may not be necessary. We must be able to cover everything in the first `if` case.